Sign providers and backoffice signs (#7325)
umbraco/UmbracoDocs: 16/umbraco-cms/customizing/back-office-signs.md

---
description: Describes how to use sign information provided in management API responses to present additional details to consumers.
---

# BackOffice Signs

> When trees, collections and items are presented in the backoffice, additional information can be displayed in the form of signs. Which signs to present will be determined by the management API response, that will be populated with a collection of `signs`.

Reviewer: ```suggestion
When trees, collections and items are presented in the backoffice, additional information can be displayed in the form of Signs. Which Signs to present will be determined by the Sign Manifest.

A Sign can utilize conditions in the same way as other Extensions or it can be bound to a Flag coming from the Manegement API.
```

> For example, a document scheduled for future publishing will have the following as part of the trees, collections and item response:

Reviewer: ```suggestion
For example, a Document scheduled for future publishing will have the following as part of the trees, collections and item response:
```

```json
  "signs": [
    {
      "alias": "Umb.ScheduledForPublish"
    }
  ],
```

> 

Reviewer: ```suggestion

A Flag can be the determinant for a Sign by declaring the `forEntityFlags` as part of its Manifest.

Example:
```
    ...
    forEntityFlags: "Umb.ScheduledForPublish",
    ....
```

```

---
description: Describes how to use sign information provided in management API responses to present additional details to consumers.
---

# BackOffice Signs

> When trees, collections and items are presented in the backoffice, additional information can be displayed in the form of Signs. Which Signs to present will be determined by the Sign Manifest.

Reviewer: ```suggestion
When trees, collections and items are presented in the backoffice, additional information can be displayed in the form of Signs.

A Backoffice sign is a client-side extension point that determines how each sign is displayed in the backoffice.
```

---
description: Describes how to use sign information provided in management API responses to present additional details to consumers.
---

> # BackOffice Signs

Reviewer 2: ```suggestion
# Backoffice Signs
```

When trees, collections and items are presented in the backoffice, additional information can be displayed in the form of Signs.

A Backoffice sign is a client-side extension point that determines how each sign is displayed in the backoffice.

A Sign can utilize conditions in the same way as other Extensions or it can be bound to a Flag coming from the Management API.

For example, a Document scheduled for future publishing will have a Flag defined as part of trees, collections and item response:

```json
  "flags": [
    {
      "alias": "Umb.ScheduledForPublish"
    }
  ],
```

A Flag can be the determinant for a Sign by declaring the `forEntityFlags` as part of its Manifest.

Example:
```json
...
forEntityFlags: "Umb.ScheduledForPublish",
...
```

Using this binding lets the server determine which signs are present in the response via the registered collection of [flag providers](../extending/flag-providers.md).

## Displaying a Sign

{% hint style="info" %}

> The client extension for backoffice signs is due in Umbraco 16.4 / 17.0.

Reviewer 2: ```suggestion
The client extension for backoffice signs will be available in Umbraco 16.4 / 17.0.
```
